### PR TITLE
fix: error messages of invalid path patterns

### PR DESCRIPTION
**Pull request by jjtang1985**
There are two failing generation due to the special paths:
- /somePath/{param1}:{param2}, where `:` is used as delimiter between multiple parameters
- /somePath/value?param={date}-{time}, where `?` and query parameters are shown in the path

I added a validation step and improved the error message.

### CODE REVIEW

File: packages/openapi-generator/src/parser/operation.ts [hunk 1]
```diff
@@ -72,13 +72,33 @@ function isPlaceholder(pathPart: string): boolean {
   return /^\{.+\}$/.test(pathPart);
 }
 
+function validatePlaceholder(pathPart: string): boolean {
```
Comment (FrankEssenberger) [inline packages/openapi-generator/src/parser/operation.ts:75]: Could you add some small statement what this method checks with an example. The regex is so hard to read.

File: packages/openapi-generator/src/parser/operation.ts [hunk 2]
```diff
@@ -72,13 +72,33 @@ function isPlaceholder(pathPart: string): boolean {
...
+  return /^\{[^{}]+\}$/.test(pathPart);
+}
+
+function validatePathPattern(
```
Comment (FrankEssenberger) [inline packages/openapi-generator/src/parser/operation.ts:79]: General: The methods are called validateXYZ and give back a boolean. For boolean I like the `isSomeThing` form. For validate I expect that the method throws an exeception if not valid.

File: packages/openapi-generator/src/parser/operation.ts (hunk 1 above)
Comment (FrankEssenberger) [reply packages/openapi-generator/src/parser/operation.ts:75]: The argument is called pathPart but it is called with placeholder. Rename?

File: packages/openapi-generator/src/parser/operation.ts [hunk 3]
```diff
@@ -72,13 +72,33 @@ function isPlaceholder(pathPart: string): boolean {
...
+  pathPattern: string,
+  placeholders: string[]
+): boolean {
+  return (
+    pathPattern.includes('?') ||
+    placeholders.some(placeholder => !validatePlaceholder(placeholder))
```
Comment (FrankEssenberger) [inline packages/openapi-generator/src/parser/operation.ts:85]: Is place holder a two word thing so `placeHolder` I am not sure, perhaps you can find out.

File: packages/openapi-generator/src/parser/operation.ts (hunk 2 above)
Comment (FrankEssenberger) [reply packages/openapi-generator/src/parser/operation.ts:79]: Also here add some small comment what the method does.

**Review by FrankEssenberger — APPROVED** [review]
LGTM some minor comments on names. Please have a look.

File: packages/openapi-generator/src/parser/operation.ts (hunk 3 above)
Comment (jjtang1985) [reply packages/openapi-generator/src/parser/operation.ts:85]: [placeholder](https://dictionary.cambridge.org/dictionary/english/placeholder) is a word.

**Pull request merged**